### PR TITLE
8345403: Add more randomized tests to better cover FloatingDecimal parsing

### PR DESCRIPTION
**Pull request by rgiulietti**
`jdk.internal.math.FloatingDecimal` parsing is mostly exercised on inputs produced by `Float.toString` and `Double.toString`, but should be extended on more random inputs.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345403](https://bugs.openjdk.org/browse/JDK-8345403): Add more randomized tests to better cover FloatingDecimal parsing (**Enhancement** - P4)

### Reviewers
 * [Joe Darcy](https://openjdk.org/census#darcy) (@jddarcy - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22523/head:pull/22523` \
`$ git checkout pull/22523`

Update a local copy of the PR: \
`$ git checkout pull/22523` \
`$ git pull https://git.openjdk.org/jdk.git pull/22523/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22523`

View PR using the GUI difftool: \
`$ git pr show -t 22523`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22523.diff">https://git.openjdk.org/jdk/pull/22523.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22523#issuecomment-2515370227)
</details>

### CODE REVIEW

**Comment by rgiulietti** [comment]
All aspects of a syntactically correct input should be randomized.

File: test/jdk/jdk/internal/math/FloatingDecimal/TestRandomFloatingDecimal.java [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+    private static void appendExponent(StringBuilder sb, int e, boolean forDec) {
+        if (forDec) {
+            sb.append(RANDOM.nextBoolean() ? 'e' : 'E');
+        } else {
+            sb.append(RANDOM.nextBoolean() ? 'p' : 'P');
+        }
+        if (e < 0) {
+            sb.append('-');
+        } else if (e == 0) {
+            switch (RANDOM.nextInt(4)) {  // 50% chance of tacit sign
```
Comment (liach) [inline test/jdk/jdk/internal/math/FloatingDecimal/TestRandomFloatingDecimal.java:213]: Maybe just call `appendRandomSign(sb)` here?

File: test/jdk/jdk/internal/math/FloatingDecimal/TestRandomFloatingDecimal.java [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+        }
+    }
+
+    private static void appendHexPrefix(StringBuilder sb) {
+        /* Randomize case of x. */
+        sb.append('0').append(RANDOM.nextBoolean() ? 'x' : 'X');
+    }
+
+    private static void appendZeros(StringBuilder sb, int count) {
+        for (; count > 0; --count) {
```
Comment (liach) [inline test/jdk/jdk/internal/math/FloatingDecimal/TestRandomFloatingDecimal.java:239]: Can use `sb.repeat('0', count)`

File: test/jdk/jdk/internal/math/FloatingDecimal/TestRandomFloatingDecimal.java [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Random;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
```
Comment (liach) [inline test/jdk/jdk/internal/math/FloatingDecimal/TestRandomFloatingDecimal.java:36]:
```suggestion
/*
```
Jtreg directives is usually formatted as block comments, and is often placed between license header and package/imports.

File: test/jdk/jdk/internal/math/FloatingDecimal/TestRandomFloatingDecimal.java [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+            sb.append(randomHexDigit(0));
+        }
+    }
+
+    private static char randomHexDigit(int min) {
+        int c = Character.forDigit(RANDOM.nextInt(min, 0x10), 0x10);
+        /* Randomize letter case as well. */
+        return (char) (c >= 'a'
+                        ? RANDOM.nextBoolean() ? c : c - ('a' - 'A')
+                        : c);
```
Comment (vy) [inline test/jdk/jdk/internal/math/FloatingDecimal/TestRandomFloatingDecimal.java:263]: I guess we can also simply do

```
return (char) (RANDOM.nextBoolean() ? Character.toUpperCase(c) : Character.toLowerCase(c));
```

File: test/jdk/jdk/internal/math/FloatingDecimal/TestRandomFloatingDecimal.java (hunk 3 above)
Comment (rgiulietti) [reply test/jdk/jdk/internal/math/FloatingDecimal/TestRandomFloatingDecimal.java:36]: Copy&paste "error"

**Review by jddarcy — APPROVED** [review]

**Pull request closed without merging**